src/Interfaces/AbstractAffineMap.jl

```diff
@@ -0,0 +1,303 @@
+
+### Input
+
+- `x`  -- point/vector
+- `am` -- affine map of a convex set
+
+### Output
+
+`true` iff ``x ∈ am``.
+
+### Algorithm
+
+Note that ``x ∈ M⋅S ⊕ v`` iff ``M^{-1}⋅(x - v) ∈ S``.
+This implementation does not explicitly invert the matrix, which is why it also
+works for non-square matrices.
+
+### Examples
+
+```jldoctest
+julia> am = AffineMap([2.0 0.0; 0.0 1.0], BallInf([1., 1.], 1.), [-1.0, -1.0]);
+
+julia> [5.0, 1.0] ∈ am
+false
+
+julia> [3.0, 1.0] ∈ am
+true
+```
+
+An example with a non-square matrix:
+
+```jldoctest
+julia> B = BallInf(zeros(4), 1.);
+
+julia> M = [1. 0 0 0; 0 1 0 0]/2;
+
+julia> [0.5, 0.5] ∈ M*B
+true
+```
+"""
+function ∈(x::AbstractVector{N}, am::AbstractAffineMap{N}) where {N<:Real}
+    return matrix(am) \ (x - vector(am)) ∈ set(am)
+end
+
+"""
+    vertices_list(am::AbstractAffineMap{N};
+                  [apply_convex_hull]::Bool) where {N<:Real}
+
+Return the list of vertices of a (polyhedral) affine map.
+
+### Input
+
+- `am`                -- affine map
+- `apply_convex_hull` -- (optional, default: `true`) if `true`, apply the convex
+                         hull operation to the list of vertices transformed by
+                         the affine map
+
+### Output
+
+A list of vertices.
+
+### Algorithm
+
+This implementation computes all vertices of `X`, then transforms them through
+the affine map, i.e. `x ↦ M*x + v` for each vertex `x` of `X`. By default, the
+convex hull operation is taken before returning this list. For dimensions three
+or higher, this operation relies on the functionality through the concrete
+polyhedra library `Polyhedra.jl`.
+
+If you are not interested in taking the convex hull of the resulting vertices
+under the affine map, pass `apply_convex_hull=false` as a keyword argument.
+
+Note that we assume that the underlying set `X` is polyhedral, either concretely
+or lazily, i.e. there the function `vertices_list` should be applicable.
+"""
+function vertices_list(am::AbstractAffineMap{N};
+                       apply_convex_hull::Bool=true) where {N<:Real}
+    # for a zero linear map, the result is just the affine translation
+    A = matrix(am)
+    b = vector(am)
+    if iszero(A)
+        return [b]
+    end
+
+    # collect vertices list of the wrapped set
+    vlist_X = vertices_list(set(am))
+
+    # create resulting vertices list
+    vlist = [A * x + b for x in vlist_X]
+
+    return apply_convex_hull ? convex_hull!(vlist) : vlist
+end
+
+"""
+    constraints_list(am::AbstractAffineMap{N}) where {N<:Real}
+
+Return the list of constraints of a (polyhedral) affine map.
+
+### Input
+
+- `am` -- affine map
+
+### Output
+
+The list of constraints of the affine map.
+
+### Notes
+
+We assume that the underlying set `X` is polyhedral, i.e., offers a method
+`constraints_list(X)`.
+
+### Algorithm
+
+Falls back to the list of constraints of the translation of a lazy linear map.
+"""
+function constraints_list(am::AbstractAffineMap{N}) where {N<:Real}
```

Comment:
Similar comment to the `vertices_list`.

Reply:
This function just calls `_constraints_list_translation` which already has such a check.